waveprop/fresnel.py

```diff
@@ -32,14 +32,15 @@ def fresnel_one_step(u_in, wv, d1, dz):
 
     # coordinates
     x1, y1 = sample_points(N=[Ny, Nx], delta=d1)
+    xx1, yy1 = np.meshgrid(x1,y1)
     x2, y2 = sample_points(N=[Ny, Nx], delta=[1 / Ny / d1[0] * wv * dz, 1 / Nx / d1[1] * wv * dz])
-
+    xx2, yy2 = np.meshgrid(x2,y2)
```

Comment:
is the mesh grid needed? from what I remember x1 and y2 and row and column vectors so that broadcasting avoids the need of meshgrid.